src/Number.php

```diff
@@ -166,23 +167,61 @@ public function getIntegerRoundingMultiplier()
     /**
      * @param string $number
      *
-     * @return bool
+     * @return string
      */
-    private static function validateNumberAsInteger($number)
+    private static function parseIntegerPart($number)
     {
-        // Check if number is invalid because of integer overflow
-        $invalid = array_filter(
-            str_split($number, strlen((string) PHP_INT_MAX) - 1),
-            function ($chunk) {
-                // Leading zeros should not invalidate the chunk
-                $chunk = ltrim($chunk, '0');
+        if ('' === $number || '0' === $number) {
+            return '0';
+        }
 
-                // Allow chunks containing zeros only
-                return '' !== $chunk && false === filter_var($chunk, FILTER_VALIDATE_INT);
+        if ('-' === $number) {
+            return '-0';
+        }
+
+        $nonZero = false;
+
+        for ($position = 0, $characters = strlen($number); $position < $characters; ++$position) {
+            $digit = $number[$position];
+            if (!isset(static::$numbers[$digit]) && !($position === 0 && $digit === '-')) {
```

Comment:
Just for curiosity, what is the reason for using Yoda style? I would guess that typos are hardly imaginable with tests, code reviews and triple equal operators... isn't readability more valuable?